test/extensions/tracers/opentelemetry/opentelemetry_tracer_impl_test.cc

```diff
@@ -115,14 +114,54 @@ TEST_F(OpenTelemetryDriverTest, ParseSpanContextFromHeadersTest) {
 
   // Remove headers, then inject context into header from the span.
   request_headers.remove(OpenTelemetryConstants::get().TRACE_PARENT);
+  request_headers.remove(OpenTelemetryConstants::get().TRACE_STATE);
   span->injectContext(request_headers, nullptr);
 
   auto sampled_entry = request_headers.get(OpenTelemetryConstants::get().TRACE_PARENT);
-
   EXPECT_EQ(sampled_entry.size(), 1);
   EXPECT_EQ(
       sampled_entry[0]->value().getStringView(),
       absl::StrJoin({version, trace_id_hex, Hex::uint64ToHex(new_span_id), trace_flags}, "-"));
+
+  auto sampled_tracestate_entry = request_headers.get(OpenTelemetryConstants::get().TRACE_STATE);
+  EXPECT_EQ(sampled_tracestate_entry.size(), 1);
+  EXPECT_EQ(sampled_tracestate_entry[0]->value().getStringView(), "test=foo");
+  const std::string request_yaml = R"(
+resource_spans:
+  resource:
+    attributes:
+      key: "service.name"
+      value:
+        string_value: "unknown_service:envoy"
+  instrumentation_library_spans:
```

Comment:
Should this be `resource_spans:` ?

Reply:
I think as written it's OK, since `instrumentation_library_spans` is the child field of `resource_spans`, but now that I think about it, this should really be `scope_spans` with that [other PR](https://github.com/envoyproxy/envoy/pull/22694). This is the classic dueling PRs though, where with name change, this test would only pass once the other is submitted...

Happy to fix once the other PR is in, or if this one is merged first I can update this code in the other PR.

Reply:
Speaking of that PR :) Updated this with `scope_spans` to match the new behavior. Thanks!

Reply:
:+1: Not really related to this PR but interested in how to leverage the config at https://github.com/envoyproxy/envoy/blob/main/test/extensions/tracers/opentelemetry/opentelemetry_tracer_impl_test.cc#L337-L357. Not clear from the test code where this is configured.

Reply:
I definitely think it could use a little refactoring to be clearer (especially the bytes setting). I was currently using it to verify the final span that the Tracer exports, but so far I've only added the exact proto matching with this config for a few of the tests (you'll see `sendMessageRaw_(_, _)` in others instead of `sendMessageRaw_(Grpc::ProtoBufferEqIgnoreRepeatedFieldOrdering(request_proto), _)`).  For legibility, I think it would also be helpful to only test the specific fields for each test to help reduce the busyness of each expectation.